raster/r.patch/main.c

```diff
@@ -282,9 +317,50 @@ int main(int argc, char *argv[])
          */
         G_verbose_message(_("Creating support files for raster map <%s>..."),
                           new_name);
-        support(names, statf, nfiles, &cats, &cats_ok, &colr, &colr_ok,
-                out_type);
+
+        // try to merge all of thread_statf[] into a single statf at
+        // thread_statf[0]
+        int next_cell_stat;
+        long merge_count;
+        // for every file
+        for (int i = 0; i < nfiles; i++) {
+            // merge all thread data into thread 0
+            for (int t = 1; t < nprocs; t++) {
+                if (out_type == CELL_TYPE) {
+                    Rast_rewind_cell_stats(thread_statf[t] + i);
+                    while (Rast_next_cell_stat(&next_cell_stat, &merge_count,
+                                               thread_statf[t] + i)) {
+                        if (next_cell_stat &&
+                            !Rast_find_cell_stat(next_cell_stat, &merge_count,
+                                                 thread_statf[0] + i)) {
+                            Rast_update_cell_stats(&next_cell_stat, 1,
+                                                   thread_statf[0] + i);
+                        }
```

Comment:
I think this works fine for this case. Practically, the merged stats won't reflect the actual stats, because here you are  interested in whether the category is there and not really how many cells of that category is there, hence `Rast_update_cell_stats(&next_cell_stat, 1, ...)`  will work although you are adding just 1 cell of that category in there.